File: api/src/main/java/com/mystore/manager/api/admin/criteria/SessionLogCriteria.java
package com.mystore.manager.api.admin.criteria;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.mystore.manager.api.common.criteria.PaginationCriteria;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class SessionLogCriteria extends PaginationCriteria {
    private Integer id;
    private Integer userId;
    private Integer posId;
    private String ipAddress;
    private String loginAtFrom;
    private String loginAtTo;
}

File: ui-admin/src/app/backend/criteria/admin/session-log-criteria.ts
import { CommonCriteria } from '../commoncriteria';

export class SessionLogCriteria extends CommonCriteria {
    id: number;
    userId: number;
    posId: number;
    ipAddress: string;

    constructor() {
        super();
        this.id = null;
        this.userId = null;
        this.posId = null;
        this.ipAddress = null;
    }
}
